### PR TITLE
Mention Microsoft CHERI-RISC-V variant; gap analysis and spec strategy

### PR DESCRIPTION
**Pull request by rwatson**
Signed-off-by: Robert N. M. Watson <robert.watson@cl.cam.ac.uk>

### CODE REVIEW

File: CHARTER.md
Comment (marnovandermaas) [inline CHARTER.md]: I would hold off on mentioning specifications that have not been published yet.

Comment (marnovandermaas) [reply CHARTER.md]: Also, if we mention Microsoft's effort, do we also have to mention Arm Morello?

File: CHARTER.md
Comment (marnovandermaas) [inline CHARTER.md]: Again, maybe make this a bit more vague saying that we should consider other variations. I would be hesitant to name Microsoft's specification specifically if it has not yet been published.

**Review by marnovandermaas — CHANGES_REQUESTED** [review]

File: CHARTER.md
Comment (rwatson) [reply CHARTER.md]: Microsoft’s spec will be published in the next few weeks, and I think will represent a pretty major activity for the group. If we don’t want to have it be a charter activity, we might be better off not proceeding with the charter until they’ve released it.

While Arm Morello is obviously amazing and will influence everyone’s thinking, it isn’t a proposal to extend RISC-V. Whereas I think we might well want to standardise a microcontroller variant of CHERI-RISC-V modelled on (and validated by) Microsoft’s work.

File: CHARTER.md
Comment (rwatson) [reply CHARTER.md]: As we currently understand it, there will be two existing specifications on the table: The SRI/Cambridge specification (for 32/64-bit), and the derived Microsoft specification (for 32-bit small cores). The sets of tradeoffs in their implementations (and especially encodings) are quite different despite their both implementing the CHERI model. So I suspect in practice we’ll want to do what I’ve written above. We can not say that and just hand wave, but if the reason is that it’s not yet published, we just wait a few weeks until it is. And if it doesn’t turn up in a few weeks, then we can try again with more vague intentions. Or are there also other reasons not to specifically name the Microsoft spec that we should be considering?

File: CHARTER.md
Comment (marnovandermaas) [reply CHARTER.md]: Ok, that makes sense.

File: CHARTER.md
Comment (marnovandermaas) [reply CHARTER.md]: Ok, as long as there is a way to amend the charter in the way that you mention, then I am happy with keeping this in.

**Review by marnovandermaas — COMMENTED** [review]
Thanks for the clarifications.

**Review by marnovandermaas — APPROVED** [review]
Looks good to me, but it would be good for more people to have a look before merging.

**Pull request merged**
